contrib/babelfishpg_tds/src/backend/tds/tdsutils.c

```diff
@@ -973,7 +975,8 @@ is_babelfish_role(const char *role)
 		&& OidIsValid(bbf_msdb_guest_oid)
 		&& is_member_of_role(role_oid, bbf_master_guest_oid)
 		&& is_member_of_role(role_oid, bbf_tempdb_guest_oid)
-		&& is_member_of_role(role_oid, bbf_msdb_guest_oid))
+		&& is_member_of_role(role_oid, bbf_msdb_guest_oid)
+		&& is_member_of_role(bbf_admin_oid, role_oid))
```

Comment:
What about only checking membership against bbf_role_admin? Is it sufficient or do we still need the above checks?

Reply:
Tried, that wasn't sufficient for a case where a TSQL user having sysadmin privilege was able to create/alter a user via PG port. Since TSQL user created the PG user, it was being considered as a BBF user and didn't get dropped.